### ISSUE TITLE
Quantities in Matplotlib

### ISSUE THREAD

**Issue by aaaandi** [hgrecco/pint#1480]
While plotting of physical quantities works with matplotlib as demonstrated in [this example](https://pint.readthedocs.io/en/stable/plotting.html), I don't understand why the following does not:

```
import matplotlib.pyplot as plt
import numpy as np
import pint

ureg = pint.UnitRegistry()
ureg.setup_matplotlib(True)

y = pint.Quantity(np.linspace(0, 30), ureg.miles)
x = pint.Quantity(np.linspace(0, 5), ureg.hours)

fig, ax = plt.subplots()
ax.plot(x, y, 'tab:blue')
plt.show()
```

Also getting the registry with `ureg = pint.get_application_registry()` does not help. Any hints?

I would appreciate any insight into this library since I really like it! Thanks a lot.

**Comment by jules-ch** [comment]
You're setting up matplotlib for `ureg` registry. But using `pint.Quantity`.

replace `pint.Qunatity` with `ureg.Quantity`

**Comment by keewis** [comment]
to elaborate on that, `pint.Quantity` uses the application registry, which means that you end up not using `ureg` at all.

I would expect your code to work if you use the application registry, though. Can you post what exactly you did that failed?

**Comment by jules-ch** [comment]
Yes, I think we need to investigate, but when I insected `ids` of registry they were different.

**Comment by keewis** [comment]
remember that the application registry is a wrapper of the actual `UnitRegistry`, so you need to call `ureg.get()` to be able to compare `id`s

**Comment by jules-ch** [comment]
Oh you're right, I'll flag this as a bug. See if we can find what's going on. It might come from the wrapper or how setup_matplotlib operates.

**Comment by keewis** [comment]
well, if `setup_matplotlib` uses that: doing `ureg.attribute = True` will set that attribute on the `ApplicationRegistry` and not the wrapped `UnitRegistry`. This is already fixed and waiting for the release.

@aaaandi, can you also tell us which version of `pint` you're using?

**Comment by jules-ch** [comment]
@keewis I was able to reproduce this bug on master

**Comment by aaaandi** [comment]
Hi. Thanks for the quick answers. Using `ureg.Quantity` does indeed solve the problem (i.e., raising an `UnitStrippedWarning`)

I'm currently at version 0.18.
